packages/browser-client/src/Components/DisplayBlock.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  Container,
+  Box,
+  TabPanels,
+  TabPanel,
+  Tab,
+  TabList,
+  Tabs,
+  Table,
+  Thead,
+  Tr,
+  Th,
+  Td,
+  Text,
+  Link,
+} from '@chakra-ui/react'
+import { Block, BlockHeader, JsonHeader } from '@ethereumjs/block'
+import Common from '@ethereumjs/common'
+import { toBuffer } from 'ethereumjs-util'
+
+interface DisplayBlockProps {
+  rlpHeader: string
+  findContent: any
+}
+
+export default function DisplayBlock(props: DisplayBlockProps) {
+  const rlpHeader = props.rlpHeader
+  const chain = 'mainnet'
+  const common = new Common({ chain: chain })
+  const blockHeader = BlockHeader.fromRLPSerializedHeader(toBuffer(rlpHeader), { common: common })
+  const block = new Block(blockHeader)
+  const json = block.toJSON()
+  const header: JsonHeader = json.header!
+  const headerObj: Record<string, string | number> = {
+    baseFee: header.baseFee!,
+    baseFeePerGas: header.baseFeePerGas!,
+    bloom: parseInt(header.bloom!.toString()),
+    coinbase: header.coinbase!,
+    difficutly: header.difficulty!,
+    extraData: header.extraData!,
+    gasLimit: header.gasLimit!,
+    gasUsed: header.gasUsed!,
+    logsBloom: Number(header.logsBloom!),
+    mixHash: header.mixHash!,
+    nonce: header.nonce!,
+    number: Number(header.number!),
+    parentHash: header.parentHash!,
+    receiptTrie: header.receiptTrie!,
+    stateRoot: header.stateRoot!,
+    timestamp: new Date(parseInt(header.timestamp!)).toString(),
+    transactionsTrie: header.transactionsTrie!,
+    uncleHash: header.uncleHash!,
+  }
```

Comment:
You should really also remove this `headerObj` structure, even if you have somewhat of an implementation reason for it (the non-null assertion operator?).

Beside the fact that it looks heavily redundant, this won't work for old-style blocks pre-London not including the 1559 fee mechanism yet and therefore not having a `baseFee` and `baseFeePerGas` field. So this won't work for the chain history network to full extend yet and is therefore unnecessarily limiting the scope here.

Reply:
(otherwise: nice to see the EthereumJS libs integrated here 🙂)

Reply:
agree with holger that it should be more dynamic to be more resilient for changes from hardforks

for the returned type if you needed the non-null assertions you can try casting as [`Required<Type>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#requiredtype) so `Required<JsonHeader>`